### PR TITLE
feat(launch): noop builder

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-10907](https://wandb.atlassian.net/browse/WB-10907)
Fixes #NNNN

Description
-----------
Adds a noop builder option for users who dont want a builder

Testing
-------
How was this PR tested?

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/launch/builder/noop.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+from wandb.errors import LaunchError
+from wandb.sdk.launch.builder.abstract import AbstractBuilder
+
+from .._project_spec import EntryPoint, LaunchProject
+
+
+class NoOpBuilder(AbstractBuilder):
+
+    type: str
```
Comment (gtarpenning) [inline wandb/sdk/launch/builder/noop.py:11]: woah you can add typing in a class def like this? cool.

Comment (gtarpenning) [reply wandb/sdk/launch/builder/noop.py:11]: presumably returns when asked for __type?

**Review by gtarpenning — APPROVED** [review]

File: wandb/sdk/launch/builder/noop.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/builder/noop.py:11]: Hmm, I'm not sure how I added this as str. Should be "noop" nice catch

File: wandb/sdk/launch/builder/noop.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def __init__(self, builder_config: Dict[str, Any]) -> None:
+        self.builder_config = builder_config
+
+    def build_image(
+        self,
+        launch_project: LaunchProject,
```
Comment (jamie-rasmussen) [inline wandb/sdk/launch/builder/noop.py:18]: Do we use pylint? I would have expected (ignorable) warnings about unused arguments here.

File: tests/unit_tests_old/tests_launch/test_launch.py [hunk 3]
```diff
@@ -1542,3 +1542,41 @@ def test_launch_git_version_default_main(
...
+        default_settings=test_settings, load_settings=False
+    )
+    monkeypatch.setattr(
+        wandb.sdk.launch.launch,
+        "LAUNCH_CONFIG_FILE",
+        "./config/wandb/launch-config.yaml",
+    )
+
+    with runner.isolated_filesystem():
+        os.makedirs(os.path.expanduser("./config/wandb"))
```
Comment (jamie-rasmussen) [inline tests/unit_tests_old/tests_launch/test_launch.py:1565]: expanduser isn't doing anything here or in the following line, is it?

Comment (KyleGoyette) [reply tests/unit_tests_old/tests_launch/test_launch.py:1565]: Ah nice catch, nor is it in the line below. Fixed

File: wandb/sdk/launch/builder/noop.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/launch/builder/noop.py:18]: We don't use pylint in our linting check. For a comprehensive list of our linting, checkout: https://github.com/wandb/wandb/blob/master/.circleci/config.yml#L1089

File: tests/unit_tests_old/tests_launch/test_launch.py (hunk 3 above)
Comment (KyleGoyette) [reply tests/unit_tests_old/tests_launch/test_launch.py]: Ive also removed them from the test I grabbed them from.

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**